staging/src/k8s.io/client-go/tools/portforward/portforward.go

```diff
@@ -395,6 +403,7 @@ func (pf *PortForwarder) handleConnection(conn net.Conn, port ForwardedPort) {
 
 		// Copy from the local port to the remote side.
 		if _, err := io.Copy(dataStream, conn); err != nil && !strings.Contains(err.Error(), "use of closed network connection") {
+			drainRemoteStream()
```

Comment:
Hitting this path will try to read *from* dataStream concurrent to the goroutine above which could still be running. Shouldn't we only drain in the above goroutine (the one already reading from `dataStream`)?

Reply:
make sense, fixed.